[highs] upgrade to version 1.10.0

mathgeekcoder opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

* Removed obsolete fix-compiler.patch
* Added fix-uwp.patch to fix issue with UWP builds (fixes issue with dependency on CLI11 - !uwp) 
* Added fix-cuda.patch to avoid path issue when CUDA not enabled (this will hopefully be removed in later versions)

Fixes https://github.com/microsoft/vcpkg/issues/44542

mathgeekcoder committed:
Update highs to 1.10.0

Also, removed cmake patch (I believe it might not be needed anymore)

mathgeekcoder committed:
Updated version and SHA

mathgeekcoder committed:
Reverted fix-compiler.patch removal, but updated to only impact UWP builds

mathgeekcoder committed:
Update SHA

mathgeekcoder committed:
Fixed uwp and cuda issues

* Removed obsolete fix-compiler.patch
* Fixed issues with UWP builds, as the highs tool now has a dependency on CLI11 which does not support UWP.
* Fixed minor include path issue with CUDA when not installed

mathgeekcoder committed:
Updated SHA

LilyWangLL commented:
> * Added fix-uwp.patch to avoid issue with UWP builds (exclude the tool, which has dependency on CLI11 - !uwp)
> * Added fix-cuda.patch to avoid path issue when CUDA not enabled (this will hopefully be removed in later versions)

Could you please report these changes on upstream?

LilyWangLL commented:
The following usage passed on x64-windows.
```
highs provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(highs CONFIG REQUIRED)
  target_link_libraries(main PRIVATE highs::highs)

highs provides pkg-config modules:

  # Linear Optimization Software
  highs
```

mathgeekcoder commented:
> > * Added fix-uwp.patch to avoid issue with UWP builds (exclude the tool, which has dependency on CLI11 - !uwp)
> > * Added fix-cuda.patch to avoid path issue when CUDA not enabled (this will hopefully be removed in later versions)
> 
> Could you please report these changes on upstream?

I've reported them.  The cuda issue has been fixed in the dev branch but might need to wait until their next release to remove the patch.  Still waiting to hear back about uwp support, but the library is still useful without the CLI tool.

mathgeekcoder committed:
Improved uwp support

Before I simply excluded the highs CLI tool. With this patch, I actually fix the core problem with the CLI11 dependency.

mathgeekcoder committed:
Update sha

mathgeekcoder commented:
I've just made an update to `fix-uwp.patch` (and the portfile) that actually fixes the uwp issue rather than just excluding the tool.  

I have no idea if this would fix CLI11 to support uwp, but at least it seems to work for highs.
